shepherd-model/src/main/java/com/devsda/platform/shepherd/rabbitmq/RabbitMQOperations.java

```diff
@@ -0,0 +1,102 @@
+        Connection connection = factory.newConnection();
+        return connection;
+    }
+
+    public void gracefulShutdown(Connection connection, Channel channel)
+            throws IOException, TimeoutException {
+        channel.close();
+        connection.close();
+    }
+
+    /**
+     * Bind a queue to an exchange, with no extra arguments.
+     * @see com.rabbitmq.client.AMQP.Queue.Bind
+     * @see com.rabbitmq.client.AMQP.Queue.BindOk
+     * @param queueName the name of the queue
+     * @param exchangeName the name of the exchange
+     * @param routingKey the routing key to use for the binding
+     * @return a binding-confirm method if the binding was successfully created
+     * @throws java.io.IOException if an error is encountered
+     */
+    public void bindQueue(Channel channel, String exchangeName, String queueName, String routingKey, boolean isDurable)
+            throws IOException {
+       AMQP.Exchange.DeclareOk exchangeDeclareResponse =channel.exchangeDeclare(exchangeName, "direct", true);
+
+        channel.queueBind(queueName, exchangeName, routingKey);
+    }
+
+    public void bindQueueWithMultipleComnsumers(Channel channel, String exchangeName,
+                                                String queueName, String routingKey) throws IOException {
+        channel.exchangeDeclare(exchangeName, "direct", true);
+        channel.queueDeclare(queueName, true, false, false, null);
+        channel.queueBind(queueName, exchangeName, routingKey);
+    }
+
+    public void deleteQueue(Channel channel, String queueName, Boolean ifUnused, Boolean ifEmpty) throws IOException {
+        channel.queueDelete(queueName, ifUnused, ifEmpty);
+    }
+
+    public void purgeAllMessages(Channel channel, String queueName) throws IOException {
+        channel.queuePurge(queueName);
+    }
+
+    public void publishMessage(Channel channel, String exchangeName, String routingKey, String message) throws IOException {
+
+        if (message == null || message.equals("")) {
+            return;
+        }
+
+        byte[] messageBodyBytes = message.getBytes();
+        channel.basicPublish(exchangeName, routingKey,
+                MessageProperties.PERSISTENT_TEXT_PLAIN,
+                messageBodyBytes);
+    }
+
+    public void receiveMessage(Channel channel, String queueName) throws IOException {
+        System.out.println(" [*] Waiting for messages. To exit press CTRL+C");
+
+        GetResponse response = channel.basicGet("first_queue", false);
+        if (response == null) {
+            // No message retrieved.
+        } else {
+            AMQP.BasicProperties props = response.getProps();
+            byte[] body = response.getBody();
+            String responseMessage = new String(body);
+            System.out.println(responseMessage);
+            long deliveryTag = response.getEnvelope().getDeliveryTag();
+            System.out.println("delivery Tag : "+ deliveryTag);
```

Comment:
Remove sysout. Use logs.

Reply:
This file will not get checked in so no need